File: shared/ui/ipc/webview.protocol.events.ts
import { NotificationType } from "./webview.protocol.common";

// import { State as ContextState } from "../store/context/types";
type ContextState = any;

export const WebviewReadyNotificationType = new NotificationType<void, void>(
	"extension/view-ready"
);

export interface DidChangeActiveStreamNotification {
	streamId?: string;
}

export const DidChangeActiveStreamNotificationType = new NotificationType<
	DidChangeActiveStreamNotification,
	void
>("extension/changed-active-stream");

export interface DidOpenThreadNotification {
	streamId: string;
	threadId?: string;
}
export const DidOpenThreadNotificationType = new NotificationType<DidOpenThreadNotification, void>(
	"extension/thread-opened"
);

export interface DidCloseThreadNotification {
	threadId?: string;
}

export const DidCloseThreadNotificationType = new NotificationType<
	DidCloseThreadNotification,
	void
>("extension/thread-closed");

export interface DidChangeContextStateNotification {
	state: ContextState;
}
export const DidChangeContextStateNotificationType = new NotificationType<
	DidChangeContextStateNotification,
	void
>("extension/context-state-changed");

File: vs/src/CodeStream.VisualStudio/Models/Webview.Protocol.Events.cs
namespace CodeStream.VisualStudio.Models
{
    public class WebviewReadyNotificationType : NotificationType<EmptyRequestTypeParams>
    {
        public const string MethodName = "extension/view-ready";
        public override string Method => MethodName;
    }

    public class DidChangeActiveStreamNotification
    {
        public string StreamId { get; set; }
    }
    public class DidChangeActiveStreamNotificationType : NotificationType<DidChangeActiveStreamNotification>
    {
        public const string MethodName = "extension/changed-active-stream";
        public override string Method => MethodName;
    }

    public class DidOpenThreadNotificationType : NotificationType<EmptyRequestTypeParams>
    {
        public const string MethodName = "extension/thread-opened";
        public override string Method => MethodName;
    }

    public class DidCloseThreadNotificationType : NotificationType<EmptyRequestTypeParams>
    {
        public const string MethodName = "extension/thread-closed";
        public override string Method => MethodName;
    }

    public class DidChangeContextStateNotificationType : NotificationType<EmptyRequestTypeParams>
    {
        public const string MethodName = "extension/context-state-changed";
        public override string Method => MethodName;
    }
}
